contracts/core/SedaCoreV1.sol

```diff
@@ -141,62 +163,13 @@ contract SedaCoreV1 is
         _removePendingRequest(result.drId);
         delete _storageV1().requestDetails[result.drId];
 
-        // Fee distribution: handles three types of fees (request, result, batch)
-        // and manages refunds back to the requestor when applicable
-
-        // Amount to refund to requestor
-        uint256 refundAmount;
-
-        // Request fee distribution:
-        // - if invalid payback address, send all request fee to requestor
-        // - if valid payback address, split request fee proportionally based on gas used vs gas limit
-        if (requestDetails.requestFee > 0) {
-            address payableAddress = result.paybackAddress.length == 20
-                ? address(bytes20(result.paybackAddress))
-                : address(0);
-
-            if (payableAddress == address(0)) {
-                refundAmount += requestDetails.requestFee;
-            } else {
-                // Split request fee proportionally based on gas used vs gas limit
-                uint256 submitterFee = (result.gasUsed * requestDetails.requestFee) / requestDetails.gasLimit;
-                if (submitterFee > 0) {
-                    _transferFee(payableAddress, submitterFee);
-                    emit FeeDistributed(result.drId, payableAddress, submitterFee, ISedaCore.FeeType.REQUEST);
-                }
-                refundAmount += requestDetails.requestFee - submitterFee;
-            }
+        // If fee manager is not set, skip fee distribution
+        if (msg.value == 0 && _storageV1().feeManager == IFeeManager(address(0))) {
```

Comment:
`postResult` shouldn't be payable so I suppose `msg.value` will always be 0? And if the request had fees it should have already errored in the postRequest step so I think we don't need a check here.